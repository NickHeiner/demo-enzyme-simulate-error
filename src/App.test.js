import React from 'react';
import enzymeToJson from 'enzyme-to-json';
import {mount, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {createStore} from 'redux';
import {Provider, connect} from 'react-redux';

configure({adapter: new Adapter()});

const store = createStore(state => state);

const InnerComponent = () => <span>Successful render</span>;

class ErrorBoundary extends React.Component {
  constructor() {
    super();
    this.state = {};
  }

  componentDidCatch() {
    this.setState(() => ({didCatch: true}));
  }

  render() {
    return this.state.didCatch ? 'Error caught.' : this.props.children;
  }
}

const ConnectedErrorBoundary = connect()(ErrorBoundary);

it('expected behavior', () => {
  const wrapper = mount(
    <Provider store={store}>
      <ConnectedErrorBoundary>
        <InnerComponent />
      </ConnectedErrorBoundary>
    </Provider>
  );

  expect(enzymeToJson(wrapper.find(ErrorBoundary))).toMatchSnapshot();

  wrapper.find(InnerComponent).simulateError(new Error('Rendering failed.'));

  expect(enzymeToJson(wrapper.find(ErrorBoundary))).toMatchSnapshot();
});

it('surprising behavior', () => {
  const wrapper = mount(
    <Provider store={store}>
      <ConnectedErrorBoundary>
        <InnerComponent />
      </ConnectedErrorBoundary>
    </Provider>
  // When we include this find() call, the inner child is always rendered,
  // even after it has been made to throw an error with simulateError().
  ).find(ErrorBoundary);

  expect(enzymeToJson(wrapper)).toMatchSnapshot();

  wrapper.find(InnerComponent).simulateError(new Error('Rendering failed.'));

  expect(enzymeToJson(wrapper)).toMatchSnapshot();
});
