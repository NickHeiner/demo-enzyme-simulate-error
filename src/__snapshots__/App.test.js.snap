// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expected behavior 1`] = `
<ErrorBoundary
  dispatch={[Function]}
>
  <InnerComponent>
    <span>
      Successful render
    </span>
  </InnerComponent>
</ErrorBoundary>
`;

exports[`expected behavior 2`] = `
<ErrorBoundary
  dispatch={[Function]}
>
  Error caught.
</ErrorBoundary>
`;

exports[`surprising behavior 1`] = `
<ErrorBoundary
  dispatch={[Function]}
>
  <InnerComponent>
    <span>
      Successful render
    </span>
  </InnerComponent>
</ErrorBoundary>
`;

exports[`surprising behavior 2`] = `
<ErrorBoundary
  dispatch={[Function]}
>
  <InnerComponent>
    <span>
      Successful render
    </span>
  </InnerComponent>
</ErrorBoundary>
`;
